### PR TITLE
[W5.7][W17-2]Edric Teo Jin Kai

### PR DESCRIPTION
**Pull request by etjk**

### CODE REVIEW

**Comment by etjk** [comment]
Work in progress

**Comment by hzf1174** [comment]
Good job @etjk.
Could modify helpCommand as well as you add a new command.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -72,6 +72,11 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Saving addressbook : `save`
```
Comment (amrut-prabhu) [inline docs/UserGuide.adoc:75]: Good work for updating the User Guide!
Would be good if you could give an example and explain where the file is (and what it's called)

File: src/seedu/addressbook/commands/SaveCommand.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+public class SaveCommand extends Command {
+    public static final String COMMAND_WORD = "save";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+        + ": Saves current contact list to current directory\n"
+        + "Example: " + COMMAND_WORD + " 1";
+
+    public CommandResult execute() {
+      List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/SaveCommand.java:14]: Indentation should be 4 spaces

File: src/seedu/addressbook/commands/Command.java [hunk 3]
```diff
@@ -26,6 +28,16 @@ public Command(int targetIndex) {
     protected Command() {
     }
 
+    public static String saveAddressBook(List<? extends ReadOnlyPerson> personsDisplayed){
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/Command.java:31]: Should this method really be defined here? Or would it be better if it was in the `SaveCommand` class?

Also, is `personsDisplayed` a suitable name?

File: src/seedu/addressbook/commands/Command.java [hunk 4]
```diff
@@ -26,6 +28,16 @@ public Command(int targetIndex) {
...
+        try (PrintWriter out = new PrintWriter("ab2.txt")) {
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/Command.java:32]: Avoid using magic numbers like `"ab2.txt"`

File: src/seedu/addressbook/common/Messages.java [hunk 5]
```diff
@@ -15,4 +15,6 @@
             "java seedu.addressbook.Main [STORAGE_FILE_PATH]";
     public static final String MESSAGE_WELCOME = "Welcome to your Address Book!";
     public static final String MESSAGE_USING_STORAGE_FILE = "Using storage file : %1$s";
+    public static final String MESSAGE_SAVE_CONTACT = "Addressbook saved successfully";
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/common/Messages.java:18]: Good work for defining variables for these strings

File: test/input.txt [hunk 6]
```diff
@@ -140,6 +140,12 @@
   delete 1
   list
 
+##########################################################
+# test save command
+##########################################################
```
Comment (amrut-prabhu) [inline test/input.txt:145]: Good work for updating I/O test file

**Review by amrut-prabhu — APPROVED** [review]
Good work on the LO @etjk 
You should try to update/add JUnit tests as well

Please close the PR after reading my comments.

**Pull request closed without merging**